### DISCUSSION TITLE
Issue compiling C++ interface of the api branch

### DISCUSSION THREAD

**Discussion by PabloPiaggi** [deepmodeling/deepmd-kit#397] [Q&A] [answered]
I am trying to compile the C++ deepmd interface using the api branch and I get the following error:

```
~/Software-deepmd-kit-1.0/deepmd-kit-api/source/lib/src/cuda/tabulate.cu:1:23: fatal error: cub/cub.cuh: No such file or directory
compilation terminated.
CMake Error at deepmd_op_cuda_generated_tabulate.cu.o.release.cmake:222 (message):
  Error generating
 ~/Programs/Software-deepmd-kit-1.0/deepmd-kit-api/source/build/lib/src/cuda/CMakeFiles/deepmd_op_cuda.dir//./deepmd_op_cuda_generated_tabulate.cu.o
```

I have tried to use different versions of gcc, tensorflow and cuda. I always find the same error.

Have you seen this error before?

Thank you!

**Comment by amcadmus** [comment, marked as answer]
please clone with ` --recursive` option to clone the submodule `cub`. For example
```
git clone --recursive https://github.com/deepmodeling/deepmd-kit.git deepmd-kit
```

**Reply by PabloPiaggi** [reply]
Thanks you @amcadmus for the fast answer. I'll try it out now.

**Reply by PabloPiaggi** [reply]
I confirm that this answer solves my problem. Thanks again!
